### PR TITLE
[#873] Enable view commits option when merge group

### PR DESCRIPTION
**Pull request by jamessspanggg**
Fixes part of #873 

```
The view commits option is not available when merge group option
is checked.

Let's enable the view option when merge group, as the zoom in 
function is already working with merged ramps. Let's also fix
the regression from #1106 that removes the github icon when
merge group and filterGroupSelection is 'groupByRepos'.
```

### CODE REVIEW

File: frontend/src/summary.pug [hunk 1]
```diff
@@ -90,34 +90,38 @@
...
-          a(
-            v-bind:href="getRepoLink(repo[0])", target="_blank",
-            title="click to view group's repo"
-          )
-            i.icon-button.fab.fa-github
-        .summary-charts__title--percentile.mini-font {{ getPercentile(i) }} %
       .summary-charts__title(
-        v-if="filterGroupSelection === 'groupByAuthors'",
-        v-bind:class="{ warn: repo[0].name === '-' }"
+        v-if="filterGroupSelection === 'groupByRepos' || filterGroupSelection === 'groupByAuthors'"
```
Comment (jamessspanggg) [inline frontend/src/summary.pug]: Note that I also refactored the structure of the `.summary-charts__title`. 

Previously, `.summary-charts__title` is separated into two, each for when filterGroupSelection is `groupByAuthors` or `groupByRepos`. There is overlapping code among the two, and separating into two makes it hard to understand the dynamic parts of the code.

If this changes is not needed or should be done in another PR, do let me know and I will act accordingly.

**Comment by 0blivious** [comment]
When select merge group, all the message body in Zoom is lost. Is this intended?

**Comment by jamessspanggg** [comment]
> When select merge group, all the message body in Zoom is lost. Is this intended?

I'm unable to reproduce it, can show an example?

**Comment by 0blivious** [comment]
> I'm unable to reproduce it, can show an example?

Occurs when group by author and open zoom tab, I guess it is not relevant to your PR though. Maybe should open a new issue for this.

File: frontend/src/summary.pug [hunk 2]
```diff
@@ -90,34 +90,38 @@
...
+              v-bind:href="getRepoLink(repo[0])", target="_blank",
+              title="click to view group's repo"
+            )
+              i.icon-button.fab.fa-github
+          template(v-else-if="filterGroupSelection === 'groupByAuthors'")
+            a(
+              v-bind:href="getAuthorProfileLink(repo[0].name)", target="_blank",
+              title="click to view author's profile"
+            )
+              i.icon-button.fab.fa-github
```
Comment (eugenepeh) [inline frontend/src/summary.pug]: why are there some fix on the repo link as well?

Comment (Tejas2805) [reply frontend/src/summary.pug]: When you `merge group` **group by Author** the octocat icon links to the author's profile

Comment (Tejas2805) [reply frontend/src/summary.pug]: Change brought in PR #1017

Comment (eugenepeh) [reply frontend/src/summary.pug]: that's not the point, the point is that the scope of this PR has nothing to do with repo link

Comment (Tejas2805) [reply frontend/src/summary.pug]: Aah got it! I think its cause of the refactoring of code done along with it. Referring to @jamessspanggg comment on top.

**Comment by Tejas2805** [comment]
@jamessspanggg Can you update the PR name and commit message

File: frontend/src/summary.pug (hunk 2 above)
Comment (jamessspanggg) [reply frontend/src/summary.pug]: > Note that I also refactored the structure of the .summary-charts__title.
> 
> Previously, .summary-charts__title is separated into two, each for when filterGroupSelection is groupByAuthors or groupByRepos. There is overlapping code among the two, and adding another new icon/text to the title need to be done twice (once each for groupByAuthors and groupByRepos).
> 
> If this change is not needed or should be done in another PR, do let me know and I will act accordingly.

@eugenepeh Should I make the refactoring of code within this PR or another PR?

Comment (eugenepeh) [reply frontend/src/summary.pug]: its preferable to do it in a separate PR, given that 1 is a fix while the other is an added component on separate part of the code

Comment (jamessspanggg) [reply frontend/src/summary.pug]: Noted. Putting this on hold until the refactoring is merged.

File: frontend/src/summary.pug [hunk 3]
```diff
@@ -127,12 +127,19 @@
             v-bind:href="getRepoLink(repo[0])", target="_blank",
             title="click to view group's repo"
           )
+            i.icon-button.fab.fa-github
```
Comment (jamessspanggg) [inline frontend/src/summary.pug]: note that this line is added to fix the regression from #1106, which is missing the github icon when merge group and filterGroupSelection is 'groupByRepos'

**Comment by jamessspanggg** [comment]
This PR is ready for review

**Comment by jamessspanggg** [comment]
@fzdy1914 need ur review here

**Pull request merged**